### ISSUE TITLE
Language Injection Plugin for IntelliJ

### ISSUE THREAD

**Issue by thedrow** [edgedb/edgedb#2284]
EdgeQL looks very nice but without syntax highlighting and autocompletion it would be harder to use.
An IntelliJ plugin could be created to detect strings with EdgeQL in them and automatically set the language for the string's contents. This is called [language injection](https://www.jetbrains.com/help/idea/using-language-injections.html).
You can create custom language injection plugins which will allow us to provide syntax highlighting and more importantly autocompletion.

**Comment by haikyuu** [comment]
This is a bit related to https://github.com/edgedb/edgedb-editor-plugin/issues/20

But I think you're asking for multiple things here:
- IntelliJ syntax highliting
- IntelliJ language injection: it probably makes sense to start with already supported editors and maybe by the most popular: VS Code.
- Autocompletion: this is pending on LSP and it's post 1.0

![image](https://user-images.githubusercontent.com/8558836/109436956-a45d8080-7a22-11eb-98c7-83ab3e8a7de0.png)

**Comment by unredundant** [comment]
Any update here? Autocompletion is a nice to have, but simple syntax highlighting is really crucial

**Comment by maybephilipp** [comment]
My opinion here. I love EdgeDB idea and everything, but it has cons that make it almost non-usable:

- No syntax highlight in IntelliJ - that's a huge CON because it disables ability to do an efficient refactoring or usages lookup. Imagine searching for all usages of "user" table in a big project. I would better write raw SQL with its cons, but I would have highlighting and refactoring features in IDE.
- No query builder - compared to ORMs, EdgeDB brings back a giant problem of manual concatenating strings with dynamic WHERE and ORDER BY conditions. And another problem it adds is that these concatted conditions will miss syntax highlight since IDE will not recognise EdgeDB syntax in these little strings - so we loose refactoring and usage lookup features again.

I understand that it's a young project and it will become a lot better later, but these problems makes it non-usable in real production projects for now.

p.s. vscode or any other editor is not a solution. It's just an editor that doesn't have refactoring features almost at all.

**Comment by unredundant** [comment]
This project is not _that_ young any more, been very disappointed to be waiting so long for JVM / IDE support. 

I get that the team wants to focus on their "cloud offering" but this feels ridiculous :( 

Not like I could adopt their cloud offering before they have basic editor and language support anyway.

**Comment by maybephilipp** [comment]
True. Cloud offering is great feature, I would like to use it, but only after I can at least use the db in the project 😄

**Comment by unredundant** [comment]
JVM support is here, and IntelliJ is introducing LSP support in their next version.  https://blog.jetbrains.com/idea/2023/06/intellij-idea-2023-2-eap-7

Is 2024 the year of EdgeDB?

**Comment by Arctomachine** [comment]
https://youtrack.jetbrains.com/issue/DBE-17080/EdgeDB-support
Official response for native integration
> No plans, demand low

Averaged user opinion from this topic:
> No support - no use

So unless team makes first step, it will be endless loop of "no support because low demand because no support because low demand because ...".
Meanwhile the only response from team in this topic is 3 years ago, waiting for v1 (v5 is already out as of today).
